### PR TITLE
Add GraphCutSegment extension

### PR DESCRIPTION
**Pull request by DaphneCD**
GraphCutSegment Extension is a segmentation using graph cut and starshape algorithm.

### CODE REVIEW

File: GraphCutSegment.s4ext [hunk 1]
```diff
@@ -0,0 +1,44 @@
+#
+# First token of each non-comment line is the keyword and the rest of the line
+# (including spaces) is the value.
+# - the value can be blank
+#
+
+# This is source code manager (i.e. svn)
+scm git
+scmurl NA
```
Comment (jcfr) [inline GraphCutSegment.s4ext]: The location of the extension repository is missing. You could look at other extension description file for example. See https://github.com/Slicer/ExtensionsIndex

How did you create the description file ? Manually or using the  extension wizard ?

File: GraphCutSegment.s4ext [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+# list dependencies
+# - These should be names of other modules that have .s4ext files
+# - The dependencies will be built first
+depends     NA
+
+# Inner build directory (default is ".")
+build_subdirectory .
+
+# homepage
+homepage    https://raw.githubusercontent.com/DaphneCD/GraphCutSegmentExtension/master/CMakeLists.txt
```
Comment (jcfr) [inline GraphCutSegment.s4ext]: Home page should either be a website or a wiki page hosted on the Slicer wiki.

File: GraphCutSegment.s4ext [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+# Firstname1 Lastname1 ([SubOrg1, ]Org1), Firstname2 Lastname2 ([SubOrg2, ]Org2)
+# For example: Jane Roe (Superware), John Doe (Lab1, Nowhere), Joe Bloggs (Noware)
+contributors Daphne Chen (Western University)
+
+# Match category in the xml description of the module (where it shows up in Modules menu)
+category    Examples
+
+# url to icon (png, size 128x128 pixels)
+iconurl     https://raw.githubusercontent.com/DaphneCD/GraphCutSegmentExtension/master/GraphCutSegment.png
```
Comment (jcfr) [inline GraphCutSegment.s4ext]: :+1:

File: GraphCutSegment.s4ext [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+# Give people an idea what to expect from this code
+#  - Is it just a test or something you stand behind?
+status      
+
+# One line stating what the module does
+description This is a segment extension using graph cut algorithm.
+
+# Space separated list of urls
+screenshoturls http://www.example.com/Slicer/Extensions/GraphCutSegment/Screenshots/1.png
```
Comment (jcfr) [inline GraphCutSegment.s4ext]: Would be awesome if you could upload a screenshot and linked from here.

**Comment by jcfr** [comment]
Thanks for your contribution :+1:  

Almost done. After addressing the remarks posted above, it should be good.

File: GraphCutSegment.s4ext (hunk 1 above)
Comment (DaphneCD) [inline GraphCutSegment.s4ext]: I created the description file following this page .
http://www.slicer.org/slicerWiki/index.php/Documentation/Nightly/Developers/Tutorials/Contribute_Extension_Description_File

File: GraphCutSegment.s4ext (hunk 2 above)
Comment (DaphneCD) [inline GraphCutSegment.s4ext]: I change the home page to a website.

File: GraphCutSegment.s4ext (hunk 4 above)
Comment (DaphneCD) [inline GraphCutSegment.s4ext]: A screenshot is added.

**Comment by DaphneCD** [comment]
Hi,
I've made the changes according to your comments and submitted a new commit. I'm not sure if changes are submitted this way. Or should I close this branch and pull request and start all over again? I'm not quite familiar with Github. Hope you can understand.
Waiting for the response.

Thanks a lot.

**Comment by fedorov** [comment]
scmurl is still not initialized to a meaningful value of the repository location

**Comment by DaphneCD** [comment]
Sorry that I missed jcfr's first comment. I've added the scmurl and submitted a new commit.
Please let me know if there is further problem.
Thanks.

**Comment by fedorov** [comment]
Thanks!

Now that I could access the repository of your extension - please add the license under which your extension is distributed. See the rules regarding licenses here: http://www.slicer.org/slicerWiki/index.php/Documentation/4.4/Extensions/CatalogPolicies.

@jcfr should be back tomorrow I think so he will be able to review your submission and finalize the merge. You are getting close!

**Comment by DaphneCD** [comment]
Thanks for the review.
I've added License.txt to the repository of my extension.

**Comment by DaphneCD** [comment]
Hi,
I'm wondering if my submission has been reviewed yet. Any comment is welcomed so that I can have further change and progress.
Waiting for the response.
Many thanks!

**Comment by jcfr** [comment]
Could you also set a category different from `Examples` ? What about `Segmentation` ?

**Comment by DaphneCD** [comment]
Hi,
I've changed the category. Sorry I haven't noticed it before.
Please let me know if there any more questions.
Thanks.

**Comment by jcfr** [comment]
Great. Thanks for applying the change.

Few more questions:

1) Looking at the screenshot, I noticed the module wasn't associated with a valid icon. Seems the following file is an invalid/empty png: https://github.com/DaphneCD/GraphCutSegmentExtension/blob/master/InteractiveSeg/Resources/Icons/InteractiveSeg.png

Do you think you could add a valid icon ?

2) The module is named `InteractiveSeg`, would it make sense to name it `GraphCutInteractiveSegmenter` or something along those lines ?

**Comment by DaphneCD** [comment]
Thanks for pointing out the questions. They are solved now.
1) I've put the valid icon and updated the screenshot image.
2) The name of the module has been changed to GraphCutInteractiveSegmenter. And the updated folder and files are also uploaded.

Please check again.
Feedback is greatly appreciated.

**Comment by jcfr** [comment]
Looking good. You can now submit an new description file including the new git hash.

You have few options:
* using the wizard: http://wiki.slicer.org/slicerWiki/index.php/Documentation/Nightly/Developers/Tutorials/Contribute_Extension_Description_File_using_Extension_Wizard
* manually: http://wiki.slicer.org/slicerWiki/index.php/Documentation/Nightly/Developers/Tutorials/Contribute_Extension_Description_File#Overview

**Comment by jcfr** [comment]
Waiting you submit a new PR with a new description file, I am closing this one.

**Pull request closed without merging**